### ISSUE TITLE
After updating to 4.0.17 I get `Cannot read property 'itemContainers' of undefined`

### ISSUE THREAD

**Issue by jimmykane** [amcharts/amcharts4#798]
After updating to 4.0.17 I get `Cannot read property 'itemContainers' of undefined` 

```
TypeError: Cannot read property 'itemContainers' of undefined
    at Disposer._dispose (Legend.js:155)
    at Disposer.push../node_modules/@amcharts/amcharts4/.internal/core/utils/Disposer.js.Disposer.dispose (Disposer.js:39)
    at LegendDataItem.push../node_modules/@amcharts/amcharts4/.internal/core/Base.js.BaseObject.dispose (Base.js:167)
    at LegendDataItem.push../node_modules/@amcharts/amcharts4/.internal/core/DataItem.js.DataItem.dispose (DataItem.js:381)
    at List.js:210
    at Object.dispatch (EventDispatcher.js:381)
    at EventDispatcher.js:266
    at Module.each (Array.js:74)
    at EventDispatcher.push../node_modules/@amcharts/amcharts4/.internal/core/utils/EventDispatcher.js.EventDispatcher._eachListener (EventDispatcher.js:239)
    at EventDispatcher.push../node_modules/@amcharts/amcharts4/.internal/core/utils/EventDispatcher.js.EventDispatcher.dispatchImmediately (EventDispatcher.js:264)
    at SortedList.js:176
    at Module.eachReverse (Array.js:87)
    at OrderedListTemplate.push../node_modules/@amcharts/amcharts4/.internal/core/utils/SortedList.js.OrderedList.setAll (SortedList.js:173)
    at OrderedListTemplate.push../node_modules/@amcharts/amcharts4/.internal/core/utils/SortedList.js.OrderedList.clear (SortedList.js:192)
    at Legend.push../node_modules/@amcharts/amcharts4/.internal/core/Component.js.Component.disposeData (Component.js:660)
    at Legend.set [as data] (Component.js:790)
    at XYChart.push../node_modules/@amcharts/amcharts4/.internal/charts/types/SerialChart.js.SerialChart.feedLegend (SerialChart.js:176)
    at XYChart.push../node_modules/@amcharts/amcharts4/.internal/charts/types/SerialChart.js.SerialChart.handleSeriesAdded (SerialChart.js:160)
    at XYChart.push../node_modules/@amcharts/amcharts4/.internal/charts/types/XYChart.js.XYChart.handleSeriesAdded (XYChart.js:653)
    at SerialChart.js:122
    at Object.dispatch (EventDispatcher.js:381)
    at EventDispatcher.js:266
    at Module.each (Array.js:74)
    at EventDispatcher.push../node_modules/@amcharts/amcharts4/.internal/core/utils/EventDispatcher.js.EventDispatcher._eachListener (EventDispatcher.js:239)
    at EventDispatcher.push../node_modules/@amcharts/amcharts4/.internal/core/utils/EventDispatcher.js.EventDispatcher.dispatchImmediately (EventDispatcher.js:264)
    at ListTemplate.push../node_modules/@amcharts/amcharts4/.internal/core/utils/List.js.List.push (List.js:588)
    at EventCardChartNewComponent.push../src/app/components/cards/event/chart/event.card.chart.component.ts.EventCardChartNewComponent.createOrUpdateChartSeries (event.card.chart.component.ts:324)
    at event.card.chart.component.ts:177
    at Array.map (<anonymous>)
    at MapSubscriber.project (event.card.chart.component.ts:176)
    at MapSubscriber.push../node_modules/rxjs/_esm5/internal/operators/map.js.MapSubscriber._next (map.js:35)
    at MapSubscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next (Subscriber.js:54)
    at MapSubscriber.push../node_modules/rxjs/_esm5/internal/operators/map.js.MapSubscriber._next (map.js:41)
    at MapSubscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next (Subscriber.js:54)
    at CombineLatestSubscriber.push../node_modules/rxjs/_esm5/internal/observable/combineLatest.js.CombineLatestSubscriber.notifyNext (combineLatest.js:83)
    at InnerSubscriber.push../node_modules/rxjs/_esm5/internal/InnerSubscriber.js.InnerSubscriber._next (InnerSubscriber.js:15)
    at InnerSubscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next (Subscriber.js:54)
    at MapSubscriber.push../node_modules/rxjs/_esm5/internal/operators/map.js.MapSubscriber._next (map.js:41)
    at MapSubscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next (Subscriber.js:54)
    at angularfire2.js:49
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (zone.js:388)
    at Object.onInvoke (core.js:16156)
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (zone.js:387)
    at Zone.push../node_modules/zone.js/dist/zone.js.Zone.run (zone.js:138)
    at NgZone.push../node_modules/@angular/core/fesm5/core.js.NgZone.run (core.js:16070)
    at SafeSubscriber._next (angularfire2.js:49)
    at SafeSubscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.SafeSubscriber.__tryOrUnsub (Subscriber.js:196)
    at SafeSubscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.SafeSubscriber.next (Subscriber.js:134)
    at Subscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber._next (Subscriber.js:77)
    at Subscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next (Subscriber.js:54)
console.js:35 Error: Data fields for series "Heart Rate" are not properly defined.
    at LineSeries.push../node_modules/@amcharts/amcharts4/.internal/charts/series/XYSeries.js.XYSeries.validateData (XYSeries.js:453)
    at System.push../node_modules/@amcharts/amcharts4/.internal/core/System.js.System.update (System.js:128)
    at System.js:348
    at wrapped (raven.js:376)
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask (zone.js:421)
    at Object.onInvokeTask (core.js:16147)
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask (zone.js:420)
    at Zone.push../node_modules/zone.js/dist/zone.js.Zone.runTask (zone.js:188)
    at push../node_modules/zone.js/dist/zone.js.ZoneTask.invokeTask (zone.js:496)
    at ZoneTask.invoke (zone.js:485)
    at timer (zone.js:2054)
console.js:35 TypeError: Cannot read property 'x' of undefined
    at LineSeries.push../node_modules/@amcharts/amcharts4/.internal/charts/series/LineSeries.js.LineSeries.closeSegment (LineSeries.js:337)
    at LineSeries.push../node_modules/@amcharts/amcharts4/.internal/charts/series/LineSeries.js.LineSeries.openSegment (LineSeries.js:290)
    at LineSeries.push../node_modules/@amcharts/amcharts4/.internal/charts/series/LineSeries.js.LineSeries.validate (LineSeries.js:193)
    at System.js:253
    at Module.each (Array.js:74)
    at List.push../node_modules/@amcharts/amcharts4/.internal/core/utils/List.js.List.each (List.js:727)
    at System.js:244
    at Object.js:107
    at eachContinue (Object.js:93)
    at Module.each (Object.js:106)
    at System.push../node_modules/@amcharts/amcharts4/.internal/core/System.js.System.update (System.js:217)
    at System.js:348
    at wrapped (raven.js:376)
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask (zone.js:421)
    at Object.onInvokeTask (core.js:16147)
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask (zone.js:420)
    at Zone.push../node_modules/zone.js/dist/zone.js.Zone.runTask (zone.js:188)
    at push../node_modules/zone.js/dist/zone.js.ZoneTask.invokeTask (zone.js:496)
    at ZoneTask.invoke (zone.js:485)
    at timer (zone.js:2054)
``` 

I only updated to the last lib nothing has changed on my side 

I will add more info once I got more and try to get on .`17 `

**Comment by martynasma** [comment]
Can you share your chart config? I don't get these on none of our demos, so it must be something with the particular setup.

**Comment by dmsmaxim** [comment]
It happens when you call `chart.dispose()`
Example: https://codepen.io/dmsmaxim/pen/jXvrrB

**Comment by jimmykane** [comment]
@dmsmaxim  I am not calling the dispose at that time. 

I think I am suspecting what is going on: 

- I have a legend in an outside div (container) 
- I use events to change the height (like in the tutorial). 

I think that is the problem looking at the error : `Disposer._dispose (Legend.js:155)`

So somehow the chart tries to create some legend and then dispose it ? 

Cc @martynasma

**Comment by dmsmaxim** [comment]
@jimmykane I just have the same error. And it happens when I call `chart.dispose()`.

**Comment by jimmykane** [comment]
@dmsmaxim  ok cool. I am not saying that it cannot happen for that call. I just said that was not my case... (Maybe something else that affects us both is broken) :-D

**Comment by martynasma** [comment]
OK, thanks everyone. Was able to reproduce. We've got this.

**Comment by martynasma** [comment]
BTW, it happens only if chart that is being `dispose()`'ed has `Legend` enabled.

**Comment by jimmykane** [comment]
@martynasma  I am not disposing the chart and this happens. I don't have any line of code that calls the 
```
try {
      this.zone.runOutsideAngular(() => {
        if (this.chart) {
          this.chart.dispose();
          delete this.chart
        }
      });
    } catch (e) {
      this.logger.error('Could not destroy chart');
      // Log to Sentry
      Raven.captureException(e);
    }
```

Except if that happens internally at some point when series are being cleared or legend is created/disposed.

**Comment by martynasma** [comment]
Yeah, `chart.series.clear()` results in the same error in same place (Legend). When series are cleared they are being disposed automatically.

**Comment by martynasma** [comment]
Just released 4.0.18. Should fix the issue.

Let me know here if you are still seeing it after upgrade.
